### ISSUE TITLE
`pl.from_pandas(df, include_index=True)` silently ignores index if it is named the same way as a column

### ISSUE THREAD

**Issue by MarcoGorelli** [pola-rs/polars#15938]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
df = pd.DataFrame({'a': [1,2,3], 'b': [4,5,6]}, index=pd.Index([7,8,9], name='a'))
print(pl.from_pandas(df, include_index=True))
```

### Log output

```shell
shape: (3, 2)
┌─────┬─────┐
│ a   ┆ b   │
│ --- ┆ --- │
│ i64 ┆ i64 │
╞═════╪═════╡
│ 1   ┆ 4   │
│ 2   ┆ 5   │
│ 3   ┆ 6   │
└─────┴─────┘
```

### Issue description

the index was silently excluded

### Expected behavior

I think this should raise

### Installed versions

<details>

```
--------Version info---------
Polars:               0.20.10
Index type:           UInt32
Platform:             Linux-5.15.146.1-microsoft-standard-WSL2-x86_64-with-glibc2.35
Python:               3.11.9 (main, Apr  6 2024, 17:59:24) [GCC 11.4.0]

----Optional dependencies----
adbc_driver_manager:  <not installed>
cloudpickle:          3.0.0
connectorx:           <not installed>
deltalake:            <not installed>
fsspec:               2024.3.1
gevent:               <not installed>
hvplot:               0.9.2
matplotlib:           3.8.4
numpy:                1.26.4
openpyxl:             <not installed>
pandas:               2.2.2
pyarrow:              15.0.2
pydantic:             <not installed>
pyiceberg:            <not installed>
pyxlsb:               <not installed>
sqlalchemy:           <not installed>
xlsx2csv:             <not installed>
xlsxwriter:           <not installed>
```

</details>

**Comment by niccolopetti** [comment]
This issue can be closed after #17628

**Comment by MarcoGorelli** [comment]
yup thanks!
